File: sample-data-repository/src/main/java/nl/haploid/octowight/sample/repository/PersonDmo.java
package nl.haploid.octowight.sample.repository;

import nl.haploid.octowight.registry.data.Atom;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = PersonDmo.ATOM_TYPE, schema = "octowight")
public class PersonDmo implements Atom {

    public static final String ATOM_TYPE = "person";

    @Id
    @SequenceGenerator(name = "person_sequence", sequenceName = "octowight.person_sequence")
    @GeneratedValue(generator = "person_sequence")
    private Long id;

    @Column(name = "name")
    private String name;

    @OneToMany(mappedBy = "person")
    private List<RoleDmo> roles = new ArrayList<>();
    private String atomLocus;

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public List<RoleDmo> getRoles() {
        return this.roles;
    }

    public void setId(final Long id) {
        this.id = id;
    }

    public void setName(final String name) {
        this.name = name;
    }

    public void setRoles(final List<RoleDmo> roles) {
        this.roles = roles;
    }

    @Override
    public Long getAtomId() {
        return getId();
    }

    @Override
    public String getAtomLocus() {
        return atomLocus;
    }

    public void setAtomLocus(String atomLocus) {
        this.atomLocus = atomLocus;
    }

    @Override
    public String getAtomType() {
        return ATOM_TYPE;
    }
}

File: sample-data-repository/src/main/scala/nl/haploid/octowight/sample/repository/PersonDmo.scala
package nl.haploid.octowight.sample.repository

import java.{lang, util}
import javax.persistence._

import nl.haploid.octowight.registry.data.{Atom, Atomizable}

import scala.beans.BeanProperty

object PersonDmo {
  val AtomCategory = "person"
}

@Entity
@Table(name = "person", schema = "octowight")
class PersonDmo extends Atomizable {

  @Id
  @SequenceGenerator(name = "person_sequence", sequenceName = "octowight.person_sequence")
  @GeneratedValue(generator = "person_sequence")
  @BeanProperty var id: lang.Long = null

  @Column(name = "name")
  @BeanProperty var name: String = null

  @OneToMany(mappedBy = "person")
  @BeanProperty var roles: util.List[RoleDmo] = new util.ArrayList[RoleDmo]

  @Transient
  @BeanProperty var origin: String = null

  override def toAtom = new Atom(id, origin, RoleDmo.AtomCategory)
}
